\name{dnase2tf}
\alias{dnase2tf}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{
finds footprint candidates from the DNase-I seq data.
}
\description{
dnase2tf finds footprint candidates from the DNase-I seq data (given by
datafilepath) on the user specified regions of interest (given by
hotspotfilename).  Mappability file location (mapfiledir) and the
filename for the output file are also required.
}
\usage{
dnase2tf(datafilepath, hotspotfilename, mapfiledir, outputfilepath, 
assemseqdir = "", dftfilename = "", maxw = 30, minw = 6, z_threshold = -2,
numworker = 2, FDRs = c(0, 0.001, 0.005, 0.01, 0.05, 0.1, 0.5, 1), paired = F)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{datafilepath}{
the directory path + the file name prefix of the aligned DNase-seq reads.
The reads should be separated into several files by chromosomes and the 
file names share the common prefix and suffix in their file names.   
The names differ only by chromosome names and are assumed to be in the 
format "DATA NAME(Prefix)” + "_chr#” + ".txt".   For example, typical data 
files would be named as  "DHS_Dex_chr1.txt", "DHS_Dex_chr2.txt", etc.  
Each row in a data file represents the genomic location of a mapped sequence 
read (chromosome, start coordinate, end coordinate, strand).  Fields are 
delimited by tabs and coordinates are 1-based.  Strand is either ‘F’ or ‘R’ 
which stands for forward or reverse strand, respectively.
}
  \item{hotspotfilename}{
the file name of the genomic regions of interest in a BED-like file format.
 Each line has three tab-delimited fields, the chromosome (e.g. chr1, chrY),
 the start coordinate (0-based), and the end coordinate.  
}
  \item{mapfiledir}{
the mappability file directory of the reference genome used for alignment. 
A mappability file of a chromosome is a binary file that each nth byte
position of the file counts the frequency (values between 0 and 255) 
of the 5’-directional m-mers starting at the position of the chromosome
among the whole chromosomes in the reference genome.   Mappability 
files can be generated by a mappability mapping program developed by 
the Gerstein lab at Yale University (gerstein.org) as part of the 
PeakSeq package (http://archive.gersteinlab.org/proj/PeakSeq/Mappability_Map/Code/).
The filenames of the mappability profiles must be in the form of 
"Chromosome name"+"b.out" such as  "chr1b.out", "chr2b.out", etc.
}
  \item{outputfilepath}{
 the directory path + the file name prefix for the output files.   DNase2TF
 produces lists of footprint candidates in both BED and BEDGraph (bgr)
 file formats selected by a pre-set FDR (False Discovery Rate) between 0%% and 100%%..
}
  \item{assemseqdir}{
the directory path of the assembly sequence files (chr1.fa, chr2.fa, etc..)
which can be downloaded from the following links:
\tabular{ll}{
hg19:\tab http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz \cr
mm9:\tab http://hgdownload.soe.ucsc.edu/goldenPath/mm9/bigZips/chromFa.tar.gz \cr
}
If omitted, dinucleotide bias correction is disabled in the computation.
}
  \item{dftfilename}{
the filename of the dinucleotide frequency table (DFT) file. 
A DFT file is a tab-delimited text file containing expected and observed 
occurrences of dinucleotides measured from the user’s DNase-seq data. 
The DFT generation program (calcDFT) is provided to produce a DFT file 
from the user data.  This option is ignored if dinucleotide bias correction 
is not being used.
}
  \item{maxw}{
The maximum base pair width of footprint candidates.  '30' by default.
}
  \item{minw}{
The minimum base pair width of footprint candidates.  '6' by default.
}
  \item{z_threshold}{
z score threshold for the initial selection of footprint candidates. 
 Only footprint candidates with z score less than the threshold will be 
 selected at the initial step of the algorithm.   The threshold determined 
 by the false discovery rate will be applied to eliminate more candidates
 at a later stage.  The default value is ‘-2’.
}
  \item{numworker}{
the number of CPU cores for multi-core computation.  To assess memory burden,
it is recommended that the user try a small number first.  The default value is ‘2’.
}
\item{FDRs}{
FDR threshold(s).  The values should be between 0 and 1. 
}

\item{paired}{
paired-end sequencing.   The value is TRUE or FALSE. 
}
}
\details{
%%  ~~ If necessary, more details than the description above ~~
}
\value{
%%  ~Describe the value returned
%%  If it is a LIST, use
%%  \item{comp1 }{Description of 'comp1'}
%%  \item{comp2 }{Description of 'comp2'}
%% ...
}
\references{
%% ~put references to the literature/web site here ~
}
\author{
Songjoon Baek
}
\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{

}

% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory.
\keyword{ ~kwd1 }
\keyword{ ~kwd2 }% __ONLY ONE__ keyword per line
